experiment/README.md

```diff
@@ -0,0 +1,198 @@
+ * For each test prefix, we generate new oracles/assertion using a TOG.
+ * We use [PITest](https://pitest.org/) to report mutation score of the generated tests, and also record the number of positive/negative (failing/passing) test cases.
+
+To analyze a TOG, run
+```
+experiment.sh [tog] [target-class] [source-dir] [bin-dir]
+```
+
+For example,
+```
+experiment.sh tratto example.Stack ../path/to/src ../path/to/target/classes
+```
+
+## Research Questions
+
+An "axiomatic" oracle does not depend on specific input values or test prefixes.  For example, `methodResult != null` or `methodResult > formalParameter`.
+An example of a non-axiomatic oracle is `sum(5, 1) == 6`.
+
+Our experimental analysis addresses the following research questions:
+
+1. What is the effectiveness (precision and FPR) of Tratto for generating axiomatic oracles?
+2. How does Tratto enhance test suites when combined with tools such as EvoSuite or Randoop in terms of bug-finding ability and mutation score?
+3. Is Tratto able to recreate all (or more) axiomatic oracles generated by JDoctor? Is Tratto able to synthesize all oracles (and fix incorrect oracles) generated by TOGA?
+
+[comment]: <> (4. How does Tratto compare with ChatGPT for axiomatic oracle generation?)
+
+## Metrics
+
+To answer the above research questions, we perform two experiments: [Classification](#classification) (RQ 1, 3) and [Mutation](#mutation) (RQ 2).
+
+### Classification
+
+We say an oracle <span style="color:red">"fails"</span> the code if its corresponding test assertion fails using the current implementation.
+We say an oracle <span style="color:green">"passes"</span> the specification if the assertion *should* pass according to the specification.
+
+|                     | Code                                  | Specification                         |
+|---------------------|---------------------------------------|---------------------------------------|
+| True Positive (TP)  | <span style="color:red">Fail</span>   | <span style="color:green">Pass</span> |
+| False Positive (FP) | <span style="color:red">Fail</span>   | <span style="color:red">Fail</span>   |
+| True Negative (TN)  | <span style="color:green">Pass</span> | <span style="color:green">Pass</span> |
+| False Negative (FN) | <span style="color:green">Pass</span> | <span style="color:red">Fail</span>   |
+
+For clarification, consider the following (buggy) code snippet:
+
+```java
+/**
+ * @param a an integer
+ * @param b an integer
+ * @returns the sum of the two integer values
+ */
+int sum(int a, int b) {
+    return a - b;
+}
+```
+
+We provide an example of each class of oracle below:
+- True Positive: `sum(a, b) == (a + b)`
+- False Positive: `sum(a, b) == null`
+- True Negative: `sum(a, b) != null`
+- False Negative: `sum(a, b) == (a - b)`
+
+Intuitively, we hope to maximize True Positives and True Negatives, and minimize False Positives and False Negatives. This corresponds to a high precision and a low FPR.
```

Comment:
Typically an experiment reports precision and recall rather than precision and FPR.